independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/mapping/RequestMapper.java

```diff
@@ -138,10 +160,20 @@ private RequestMatch<T> mapFromPathMatcher(String path, PathMatcher.PathMatch<Ar
                         remaining = path.substring(matchPos);
                     }
                 }
-                return new RequestMatch(potentialMatch.template, potentialMatch.value, params, remaining);
+                if (bestMatchOnly) {
+                    return Collections
```

Comment:
List.of(..) on JDK have a better number of concrete types (2 iirc - for 0,1,2 and N elements) which tends to better perform in the caller